### PR TITLE
 Fix unexpected date of birth error when date is valid

### PR DESCRIPTION
**Pull request by haydarai**
Note: This PR also force datatable for non-members to refresh after each save to prevent bugs

### CODE REVIEW

File: views/dashboard/non-members.pug [hunk 1]
```diff
@@ -330,7 +330,7 @@ block content
             dataType: "json",
             success: function (res) {
               // if you want to reload datatable after saving, uncomment line below
-              // datatable.ajax.reload();
+              datatable.ajax.reload();
```
Comment (pionize) [inline views/dashboard/non-members.pug]: I got error: `Uncaught ReferenceError: datatable is not defined` for this line

**Comment by pionize** [comment]
when updating, I got error: CastError: Cast to date failed for value "29/01/1990" at path "dob"
I think we need to add `req.body.dob = moment(req.body.dob, 'DD/MM/YYYY, h:mm A').toDate();` in putVisitor function

File: validators/VisitorValidator.js [hunk 2]
```diff
@@ -20,14 +21,20 @@ const create = (req, res, next) => {
...
   req.assert('nric')
     .notEmpty()
     .withMessage(messages.blank('NRIC'))
     .isLength({ max: 9 })
     .withMessage(messages.maxChars('NRIC', 9));
-  req.assert('dob', messages.maxChars('Date of Birth', 21)).validateAge(21);
+  req.assert('dob', messages.minAge('Date of Birth', 21)).validateAge(21);
+
+  // Let's set the property to null if it's not a valid date.
+  if (!moment(req.body.membershipExpiry).isValid()) req.body.membershipExpiry = null;
```
Comment (pionize) [inline validators/VisitorValidator.js]: for this, I think we also need to add the format 'ddd MMM DD YYYY HH:mm:ss ZZ'
It always return error in my local

**Comment by haydarai** [comment]
@pionize can you help me fix this bug?

**Comment by pionize** [comment]
ok, I have fix the bug
please review the code

**Comment by haydarai** [comment]
@pionize awesome work!

**Pull request merged**